misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -62,6 +70,10 @@ internal class SqsJobConsumer @Inject internal constructor(
     private val running = AtomicBoolean(true)
 
     override fun run() {
+      // Don't call handlers until all services are ready, otherwise handlers will crash because the
+      // services they might need (databases, etc.) won't be ready.
+      serviceManagerProvider.get().awaitHealthy()
```

Comment:
nit: blocking here is going to screw with the timing metrics for the job.

Reply:
Timing metrics for the job happen below. This is just the runloop for the entire job processing thread.